pkgs/top-level/all-packages.nix

```diff
@@ -39314,12 +39314,10 @@ with pkgs;
   or-tools = callPackage ../development/libraries/science/math/or-tools {
     stdenv = if stdenv.isDarwin then overrideSDK stdenv "11.0" else stdenv;
     python = python3;
-    protobuf = protobuf_21;
-    # or-tools builds with -std=c++20, so abseil-cpp must
-    # also be built that way
-    abseil-cpp = abseil-cpp_202206.override {
+    protobuf = protobuf_23;
+    abseil-cpp = abseil-cpp.override {
```

Comment:
OR-Tools [requires a specific version of abseil-cpp](https://github.com/google/or-tools/blob/v9.7/Dependencies.txt#L2), and historically it almost always fails to build with versions newer than the specified one, so we should still pin this to `abseil-cpp_202301`.